packages/xorg_libs.rb

```diff
@@ -0,0 +1,272 @@
+  
+  $xorg_lib_source_name.push("libXrender-0.9.10.tar.gz")
+  $sha256sum_source.push("770527cce42500790433df84ec3521e8bf095dfe5079454a92236494ab296adf")
+  
+  $xorg_lib_source_name.push("libXcursor-1.1.15.tar.gz")
+  $sha256sum_source.push("449befea2b11dde58ba3323b2c1ec30550013bd84d80501eb56d0048e62251a1")
+  
+  
+  
+  $xorg_lib_source_name.push("libXdamage-1.1.4.tar.gz")
+  $sha256sum_source.push("4bb3e9d917f5f593df2277d452926ee6ad96de7b7cd1017cbcf4579fe5d3442b")
+  
+  $xorg_lib_source_name.push("libfontenc-1.1.3.tar.gz")
+  $sha256sum_source.push("6fba26760ca8d5045f2b52ddf641c12cedc19ee30939c6478162b7db8b6220fb")
+  
+  $xorg_lib_source_name.push("libXfont2-2.0.3.tar.gz")
+  $sha256sum_source.push("a4b761a37528353a2b83dba364d7c1fd6aef2d554a1a019815f24f7f8866890e")
+  
+  $xorg_lib_source_name.push("libXft-2.3.2.tar.gz")
+  $sha256sum_source.push("26cdddcc70b187833cbe9dc54df1864ba4c03a7175b2ca9276de9f05dce74507")
+  
+  $xorg_lib_source_name.push("libXi-1.7.9.tar.gz")
+  $sha256sum_source.push("463cc5370191404bc0f8a450fdbf6d9159efbbf274e5e0f427a60191fed9cf4b")
+  
+  
+  
+  $xorg_lib_source_name.push("libXinerama-1.1.3.tar.gz")
+  $sha256sum_source.push("0ba243222ae5aba4c6a3d7a394c32c8b69220a6872dbb00b7abae8753aca9a44")
+  
+  $xorg_lib_source_name.push("libXrandr-1.5.1.tar.gz")
+  $sha256sum_source.push("2baa7fb3eca78fe7e11a09b373ba898b717f7eeba4a4bfd68187e04b4789b0d3")
+  
+  $xorg_lib_source_name.push("libXres-1.2.0.tar.gz")
+  $sha256sum_source.push("5b62feee09f276d74054787df030fceb41034de84174abec6d81c591145e043a")
+  
+  $xorg_lib_source_name.push("libXtst-1.2.3.tar.gz")
+  $sha256sum_source.push("a0c83acce02d4923018c744662cb28eb0dbbc33b4adc027726879ccf68fbc2c2")
+  
+  $xorg_lib_source_name.push("libXv-1.0.11.tar.gz")
+  $sha256sum_source.push("c4112532889b210e21cf05f46f0f2f8354ff7e1b58061e12d7a76c95c0d47bb1")
+  
+  
+  
+  $xorg_lib_source_name.push("libXvMC-1.0.10.tar.gz")
+  $sha256sum_source.push("d8306f71c798d10409bb181b747c2644e1d60c05773c742c12304ab5aa5c8436")
+  
+  $xorg_lib_source_name.push("libXxf86dga-1.1.4.tar.gz")
+  $sha256sum_source.push("e6361620a15ceba666901ca8423e8be0c6ed0271a7088742009160349173766b")
+  
+  $xorg_lib_source_name.push("libXxf86vm-1.1.4.tar.gz")
+  $sha256sum_source.push("5108553c378a25688dcb57dca383664c36e293d60b1505815f67980ba9318a99")
+  
+  $xorg_lib_source_name.push("libdmx-1.1.3.tar.gz")
+  $sha256sum_source.push("c4b24d7e13e5a67ead7a18f0b4cc9b7b5363c9d04cd01b83b5122ff92b3b4996")
+  
+  $xorg_lib_source_name.push("libpciaccess-0.14.tar.gz")
+  $sha256sum_source.push("8d86e64893917be3dfb1c5e837888d1275399c818783474002203d751312b03c")
+  
+  
+  
+  $xorg_lib_source_name.push("libxkbfile-1.0.9.tar.gz")
+  $sha256sum_source.push("95df50570f38e720fb79976f603761ae6eff761613eb56f258c3cb6bab4fd5e3")
+  
+  $xorg_lib_source_name.push("libxshmfence-1.3.tar.gz")
+  $sha256sum_source.push("7eb3d46ad91bab444f121d475b11b39273142d090f7e9ac43e6a87f4ff5f902c")
```

Comment:
How are we going to manage these "dependency" versions when individual libraries are updated?

Reply:
I'm thinking it might be better to just use the existing crew packages as dependencies.

Reply:
The dependencies are automatically handled according to installation procedure.
1) Follow the compilation steps -> build code -> install to host
2) Build related packages (DESTDIR)
3) Remove installed packages on host.

Reply:
This makes the packaging system more modular so other packages can use the same dependencies.

Reply:
We only need to see it as one package. The internal dependency should be safely handled if the package can be sucessfully built.

Reply:
The dependency lies mainly on xorg-related source code.

Reply:
So basically this is a static build of all the libraries?

Reply:
Refer to http://www.linuxfromscratch.org/blfs/view/svn/x/x7lib.html

Reply:
Yes. Build all xorg-related libraries.

Reply:
Currently we have 32 individual xorg libraries.

Reply:
Most of these packages are really old. I think only a few of them are still in active status.

Reply:
The dependency internally is not a big problem. Once the source code updated, we should always check the build procedure. It's true for all packages. Not for this bundle one.

Reply:
Reason to use weston and/or wayland instead?  I need to get back to my day job but think about this a bit more and we can discuss it later.  I don't think we will need the kitchen sink for GUI stuff.

Reply:
There are not many software using wayland. Ubuntu 18.04 switch back to xorg